### PR TITLE
[inductur] benchmark epilogue fused matmul template

### PR DESCRIPTION
**Pull request by shunting314**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114809

Want to be a able to benchmark epilogue fused triton matmul kernel for a couple of reasons
1. @eellison  found that certain TB models (resnet50, resnet152, moco) fails sometimes in maxautotune mode on the dashboard. The issue is quite hard to repro due to flakiness. The issue only get triggered when certain triton config for certain epilogue fused kernel get picked. (disable epilogue fusion bypass the issue) It would be nice if we can have a runnable script that directly run that kernel to ease further debugging
2. this is a necessary piece to do benchmark fusion for triton matmul kernels. cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler  for this

Example runnable kernel script: https://gist.github.com/shunting314/00bdbc1b6b46bfa73d1389d8f40cd669

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/config.py [hunk 1]
```diff
@@ -167,6 +167,16 @@
...
 
+# Be able to skip gemm so that
+# 1. it's more stable to test triton matmul template in unit test. Otherwise we rely
+#    on the triton template out performing the aten kernel to properly test the
+#    triton template
+# 2. make it easier to repro some max-autotune issue. E.g., we found some max
+#    autotune issue very flaky to repro because the bad triton config get picked
+#    with certain probability. Skipping aten gemm and reducing the config pool
+#    can greatly help reproducing those issues.
+skip_aten_gemm = os.environ.get("TORCHINDUCTOR_SKIP_ATEN_GEMM") == "1"
```
Comment (eellison) [inline torch/_inductor/config.py:178]: You can get this same functionality through `max_autotune_gemm_backends`

File: torch/_inductor/codegen/triton.py [hunk 2]
```diff
@@ -857,6 +858,9 @@ def simplify_indexing(index: sympy.Expr):
 
         self.simplify_indexing = simplify_indexing
 
+    def need_numel_args(self):
```
Comment (eellison) [inline torch/_inductor/codegen/triton.py:861]: would you add docstring for this ?

**Review by eellison — APPROVED** [review]
Looks good but could we reuse `max_autotune_gemm_backends` instead of adding another config option ?

File: torch/_inductor/config.py (hunk 1 above)
Comment (shunting314) [reply torch/_inductor/config.py:178]: Didn't realize I can do that earlier:

```
def use_aten_gemm_kernels():
    return (
        not use_max_autotune() or _use_autotune_backend("ATEN")
    ) and not config.skip_aten_gemm

```

But taking another look, yes, I can avoid adding an extra config.

**Comment by shunting314** [comment]
@pytorchbot merge -i

Dalle model model timeout should not be related.

**Pull request closed without merging**